pkg/operator/staticpod/controller/revision/revision_controller.go

```diff
@@ -143,19 +146,20 @@ func nameFor(name string, revision int32) string {
 }
 
 // isLatestRevisionCurrent returns whether the latest revision is up to date and an optional reason
-func (c RevisionController) isLatestRevisionCurrent(revision int32) (bool, string) {
+func (c RevisionController) isLatestRevisionCurrent(revision int32) (bool, string, string) {
 	configChanges := []string{}
+	configChangesDetailed := []string{}
 	for _, cm := range c.configMaps {
 		requiredData := map[string]string{}
 		existingData := map[string]string{}
 
 		required, err := c.configMapGetter.ConfigMaps(c.targetNamespace).Get(cm.Name, metav1.GetOptions{})
 		if apierrors.IsNotFound(err) && !cm.Optional {
-			return false, err.Error()
+			return false, err.Error(), ""
```

Comment:
so there is a chance that the annotation is empty?

Reply:
i believe in this case we won't create the installer pod at all, so that value is not used.